lib/floe/workflow/context.rb

```diff
@@ -21,10 +21,46 @@ def execution
         @context["Execution"]
       end
 
+      def start?
+        execution.key?("StartTime")
+      end
+
+      def end?
+        execution.key?("EndTime")
+      end
+
       def state
         @context["State"]
       end
 
+      def input
+        state["Input"]
+      end
+
+      def output
+        state["Output"]
+      end
+
+      def state_name
+        state["Name"]
+      end
+
+      def next_state
+        state["NextState"]
+      end
+
+      def status
+        if !start?
+          "pending"
+        elsif !end?
+          "running"
+        elsif state["Error"]
+          "failure"
+        else
+          "success"
```

Comment:
Hm if one step has finished (and thus has set execution["EndTime"] for _that_ state) will this report that the whole workflow is status "success" ?

Should we be clearing the execution Start and End times after the state has run?

Reply:
Ah nevermind execution["EndTime"] is only if there is no next state